arduino-core/src/cc/arduino/contributions/DownloadableContributionsDownloader.java

```diff
@@ -215,6 +214,12 @@ public boolean checkSignature(MultiStepProgress progress, URL signatureUrl, Prog
       // Download signature
       download(signatureUrl, packageIndexSignatureTemp, progress, statusText, progressListener, true);
 
+      if (skipVerification) {
+        log.info("Allowing insecure packages because allow_insecure_packages is set to true in preferences.txt" +
+          " but the signature was download");
+        return true;
```

Comment:
Returning `true` here will not update the `.sig` file if the signature is good.
IMHO we should just do everything as normal here (and do not consider the skipVerification preference).